backend/Dockerfile

```diff
@@ -15,10 +15,10 @@ COPY .env.template .env
 CMD ["yarn", "run", "start"]
 
 FROM base as builder
-RUN yarn install --frozen-lockfile --non-interactive
+RUN yarn install --production=false --frozen-lockfile --non-interactive
 COPY . .
 RUN cp .env.template .env
-RUN yarn run build
+RUN NODE_ENV=production yarn run build
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-09-11T06:32:08Z" title="Wednesday, September 11th 2019, 8:32:08 am +02:00">Sep 11, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -15,10 +15,10 @@ COPY .env.template .env
 CMD ["yarn", "run", "start"]
 
 FROM base as builder
-RUN yarn install --frozen-lockfile --non-interactive
+RUN yarn install --production=false --frozen-lockfile --non-interactive
 COPY . .
 RUN cp .env.template .env
-RUN yarn run build
+RUN NODE_ENV=production yarn run build
```


this is a bit confusing for me.. I saw your commit message "webapp was built with `NODE_ENV=production` but started with
`NODE_ENV=development`"
but it seems that now you are doing the opposite, oder? Line 18 says don't install production dependencies... but build it in a production environment??

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-09-11T09:20:49Z" title="Wednesday, September 11th 2019, 11:20:49 am +02:00">Sep 11, 2019</time>_

---


In `nuxt.config.js` our webapp is configured to build for "modern" browsers if `NODE_ENV == production`. If you later `yarn run start` it will complain if the modern build is missing. Yet, you must not to set `ENV NODE_ENV=production` in the docker image, because that will confuse `yarn run test`.

It took me some failing builds to figure that out.

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-09-11T09:24:25Z" title="Wednesday, September 11th 2019, 11:24:25 am +02:00">Sep 11, 2019</time>_

---


I think for `backend` it's indeed unnecessary to build in production. However it also does not hurt..